src/hooks/usePost.ts

```diff
@@ -0,0 +1,56 @@
+import { Dispatch, useCallback, useState } from 'react';
+import { addTodo } from '../api/todos';
+import { Todo } from '../types/Todo';
+import { ErrorType } from '../typedefs';
+
+interface Options {
+  setTodos: Dispatch<React.SetStateAction<Todo[]>>,
+  USER_ID: number,
+  showError: (error: ErrorType) => void
+}
+
+export const usePost = (options: Options) => {
+  const { setTodos, USER_ID, showError } = options;
+
+  const [tempTodo, setTempTodo] = useState<Todo | null>(null);
+  const [unableField, setUnableField] = useState(false);
+
+  const addNewTodo = useCallback(
+    async (title: string) => {
+      if (!title.trim()) {
+        showError(ErrorType.TITLE);
+
+        return;
+      }
+
+      const newTodo = {
+        title,
```

Comment:
Why do we use the `trim' method for checking, but not for creating a new case?

Reply:
Fixed